### ISSUE TITLE
[New build - DEFAULT] 20w22c: Weapons Improvements.

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2228]
More hotfixes and some improvements to the new handheld weapon feature.

Changes and new features:
 - Can equip a weapon on the ground.
 - Can drop a weapon to the ground.
 - Can load a weapon when opening the item stand with no item.

Bug Fixes:
#2213 Weapon self hit.
#2132 OR gate not updating on simultaneous query.
#2212 Player stance animation issue.
#2203 Player bounces when close to a handheld weapon.
Weapon Collision points offset.
Weapon primary fire fix.
Weapon item stand item swap fix.

**Comment by tsunamayo** [comment]
Build is live! Thanks to all the testers!

**Comment by hl2shadowslayer** [comment]
We can drop weapons now? How do we do that?

**Comment by tsunamayo** [comment]
you just press tab and drag and drop the item icon!

**Comment by Battlepixel** [comment]
The weapons are working again. Thank you for taking care of things so quickly. Unfortunately, the logic does not work again. Since the logic we use is very complicated, it will take a while before I find the time to locate the error. The error has been occurring for 1-2 weeks, it would help if you could say whether you worked on certain logic functions during this time and if so, which ones?

Regardless, I have sent you an example of the circuit with the blueprint "Hyäne 24".
If you investigate the matter yourself, the following information:

- The circuit ensures that: 
the landing flaps open and then the landing gear extends when the button is pressed for the first time 
- the landing gear retracts and then the landing flaps close the next time the button is pressed .
 -  Furthermore, the circuit is designed in such a way that errors do not occur either by holding the button or by clicking very quickly.

**Comment by tsunamayo** [comment]
yes I change the maximum input per frame to one, and since that broke multi-query into a logic gate I added a delay of one frame to pass their information down. The whole gets a bit messy, if you tells me what your circuit is made off I might pinpoint the issue.

**Comment by stratometal** [comment]
Hello. I find that sometimes a mechanical block sounds stay on a loop, it seems to happen to me more often when I use the slider to move a small rail, though it has happened with hinges controlled by other logic blocks as well.

On a second note, is there a way to control the ride height of a hovercraft using a slider or other parameters in the hovercraft computer?

**Comment by Battlepixel** [comment]
I found the mistake. It is not a bug, so sry for the false statement regarding the logic, since one of the newer patches, the threshold values ​​of the query gates can be set with decimal places. As a result, the values ​​stored in the blueprint have been transformed. The problem was solved by re-setting the query gates.
![Fahrwerk-Schaltung](https://user-images.githubusercontent.com/43222403/83080979-e31f6e00-a07f-11ea-9edf-98a938e39226.jpg)







> Hello. I find that sometimes a mechanical block sounds stay on a loop, it seems to happen to me more often when I use the slider to move a small rail, though it has happened with hinges controlled by other logic blocks as well.

I noticed the same thing.

**Comment by tsunamayo** [comment]
> Hello. I find that sometimes a mechanical block sounds stay on a loop, it seems to happen to me more often when I use the slider to move a small rail, though it has happened with hinges controlled by other logic blocks as well.
> 
> On a second note, is there a way to control the ride height of a hovercraft using a slider or other parameters in the hovercraft computer?

Not yet, put it is planned!

**Comment by tsunamayo** [comment]
@Battlepixel damn thats a complex circuit, but I see you are on top of your game! You said you made it resistant to double click or short click - could you explain that? I feel the logic system should handle that on its own without added complexity.

**Comment by Battlepixel** [comment]
![Skywanderers_Landing_Gear_V2](https://user-images.githubusercontent.com/43222403/83140121-d5a0cd00-a0ed-11ea-92e6-9ae682f8c926.jpg)

- the pulse generator (I) ensures that the self-resetting switch (II) resets even when the control button is held down. I have only marked one of the 4 built-in pulse generators to illustrate their function.
- the interlock (III) prevents another signal from the control button from being transmitted as long as the chassis is not fully retracted or extended.
